cpp/src/arrow/csv/reader.cc

```diff
@@ -546,6 +719,8 @@ class ReaderMixin {
     return ParseResult{std::move(parser), static_cast<int64_t>(parsed_size)};
   }
 
+  friend class HeaderParsingOperator;
```

Comment:
What is this class?

Reply:
Leftover from a previous implementation.  Removed.